src/include/common/object_pool.h

```diff
@@ -92,13 +52,15 @@ class ObjectPool {
 
   /**
    * Returns a piece of memory to hold an object of T.
-   * @throw NoMoreObjectException if the object pool fails to fetch memory.
-   * @throw AllocatorFailureException if the allocator fails to return a valid memory address.
+   * @throw std::length_error if the object pool has reached the limit of how many objects it may hand out.
```

Comment:
I am not sure this is the right exception to use.

Reply:
Hmm, do you want to go back to our own-defined exception then?  According to [this](https://en.cppreference.com/w/cpp/error/length_error), length_error "reports errors that result from attempts to exceed implementation defined length limits for some object."

Reply:
It's easy enough to go back to our custom exceptions so I'll just do that.